pandas/_libs/tslibs/offsets.pyx

```diff
@@ -212,13 +212,31 @@ def _validate_business_time(t_input):
             raise ValueError("time data must match '%H:%M' format")
     elif isinstance(t_input, dt_time):
         if t_input.second != 0 or t_input.microsecond != 0:
-            raise ValueError(
-                "time data must be specified only with hour and minute")
+            raise ValueError("time data must be specified only with "
+                             "hour and minute")
         return t_input
     else:
         raise ValueError("time data must be string or datetime.time")
 
 
+def _validate_n(n):
+    if n == 0 or not isinstance(n, int):
+        raise ValueError("N cannot be 0")
+
+
+def _validate_month(month):
```

Comment:
is there a reason that these are not instance methods of BaseOffset?

Reply:
My default is to not make something a method if it doesn't require `self`.

Reply:
this is not the current style. agree that these are somewhat of an edge case, but don't change just for the sake of change or its 'your style', prefer to keep the existing code style unless good reason to change.

Reply:
OK.  The fix to the error message in validate_weekday is probably worthwhile.  Should I revert the others?

Reply:
I like them in cython, but these make more sense as methods I think as they *do* use ``self`` at least for some)

Reply:
> as they do use self at least for some

None of the ones implemented here.

Reply:
right but the point is you don't need the allow_none arg if you simply make this a method (for the weekday one)

Reply:
How does making it a method change whether `allow_none` is needed?